rocAL/source/augmentations/effects_augmentations/node_normalize.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "node_normalize.h"
+
+#include <vx_ext_rpp.h>
+
+#include "exception.h"
+
+NormalizeNode::NormalizeNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) : Node(inputs, outputs) {}
+
+void NormalizeNode::create_node() {
+    if (_node)
+        return;
+
+    _compute_mean = _mean.size() ? 0 : 1;
+    _compute_stddev = _std_dev.size() ? 0 : 1;
+
+    int mean_stddev_array_size = 1;
+    auto nDim = _inputs[0]->info().num_of_dims() - 1;
+    uint axis[nDim];
+    for (unsigned i = 0; i < _batch_size; i++) {
+        int totalElements = 1;
+        unsigned *tensor_shape = _inputs[0]->info().roi()[i].end;
+        for (uint j = 0; j < nDim; j++) {
+            axis[j] = ((_axis_mask & (int)(pow(2, j))) >= 1) ? 1 : 0;
+            totalElements *= !axis[j] ? tensor_shape[j] : 1;
+        }
+        mean_stddev_array_size = std::max(mean_stddev_array_size, totalElements);
+    }
+    std::vector<float> mean_vec, stddev_vec;
+    mean_vec.resize(_batch_size * mean_stddev_array_size, 0);
+    stddev_vec.resize(_batch_size * mean_stddev_array_size, 0);
+
+    if (!_compute_mean && !_compute_stddev) {
+        for (uint i = 0; i < _batch_size; i++) {
+            for (int j = 0; j < mean_stddev_array_size; j += _mean.size()) {
```

Comment:
Can we add a check if mean and std dev is of the same size

Reply:
Yes
Better to add a validation check verifying if mean and std passed is of same size